stix_shifter_modules/elastic_ecs/stix_translation/json/stix_2_1/to_stix_map.json

```diff
@@ -201,60 +228,87 @@
       "key": "x-ecs-client.address",
       "object": "client"
     },
-    "domain": {
-      "key": "x-ecs-client.domain",
-      "object": "client"
-    },
-    "nat": {
-      "ip": {
-        "key": "x-ecs-client.nat_ip",
-        "object": "client"
+    "domain": [
+      {
+        "key": "domain-name.value",
+        "object": "client_host_domain"
       },
+      {
+        "key": "domain-name.resolves_to_refs",
+        "object": "client_host_domain",
+        "references": ["src_ip", "client_nat"]
+      }
+    ],
+    "nat": {
+      "ip": [
+        {
+          "key": "ipv4-addr.value",
+          "object": "client_nat"
+        },
+        {
+          "key": "ipv6-addr.value",
+          "object": "client_nat"
+        }
+      ],
       "port": {
-        "key": "x-ecs-client.nat_port",
-        "object": "client"
+        "key": "network-traffic.src_port",
+        "object": "client_nat_port"
       }
     },
-    "registered_domain": {
-      "key": "x-ecs-client.registered_domain",
-      "object": "client"
-    },
-    "top_level_domain": {
-      "key": "x-ecs-client.top_level_domain",
-      "object": "client"
-    },
+    "registered_domain": [
+      {
+        "key": "domain-name.value",
+        "object": "client_host_registered_domain"
+      },
+      {
+        "key": "domain-name.resolves_to_refs",
+        "object": "client_host_registered_domain",
+        "references": ["src_ip", "client_nat"]
+      }
+    ],
+    "top_level_domain": [
+      {
+        "key": "domain-name.value",
+        "object": "client_host_top_level_domain"
+      },
+      {
+        "key": "domain-name.resolves_to_refs",
+        "object": "client_host_top_level_domain",
+        "references": ["src_ip", "client_nat"]
+      }
+    ],
     "geo": {
       "city_name": {
-        "key": "x-ecs-client.geo_city_name",
-        "object": "client"
+        "key": "x-ecs-geo.city_name",
```

Comment:
I wonder if we should just create a generic custom geography object instead of tying it to ecs. Someone else is currently working on the mappings for the log analytics connector, and there is also a need there to capture geo information.

Reply:
I tend to agree. There was a [comment](https://github.com/opencybersecurityalliance/stix-shifter/issues/1409#issuecomment-1491412937) to that effect too. We can have it mimic the properties of the [Location SDO](https://docs.oasis-open.org/cti/stix/v2.1/os/stix-v2.1-os.html#_th8nitr8jb4k) to achieve genericity and completeness.